webview.h

```diff
@@ -911,6 +939,25 @@ class cocoa_wkwebview_engine {
     objc_registerClassPair(cls);
     return objc::msg_send<id>((id)cls, "new"_sel);
   }
+  id create_webkit_navigation_delegate() {
+    auto cls = objc_allocateClassPair((Class) "NSObject"_cls,
+                                      "WebkitNavigationDelegate", 0);
+    class_addProtocol(cls, objc_getProtocol("WKNavigationDelegate"));
+    class_addMethod(cls, "webView:didFinishNavigation:"_sel,
+                    (IMP)(+[](id delegate, SEL sel, id webview, id navigation) {
+                      auto w = get_associated_webview(delegate);
+                      auto url = objc::msg_send<id>(webview, "URL"_sel);
+                      auto nstr = objc::msg_send<id>(url, "absoluteString"_sel);
+                      auto str = objc::msg_send<char *>(nstr, "UTF8String"_sel);
+                      w->m_navigateCallback(str, w->m_navigateCallbackArg);
+                    }),
+                    "v@:@");
+    objc_registerClassPair(cls);
```

Comment:
A fix is not required for this PR since the fixes in #1005 aren't merged yet, but I'll just leave a note: Will crash when attempting to register the same class a second time.